akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala

```diff
@@ -307,20 +309,16 @@ class Http2ServerSpec extends AkkaSpecWithMaterializer("""
       // Reproducing https://github.com/akka/akka-http/issues/2957
       "close the stream when we receive a RST after we have half-closed ourselves as well" in new WaitingForRequestData {
         // Client sends the request, but doesn't close the stream yet. This is a bit weird, but it's whet grpcurl does ;)
-        sendHEADERS(streamId = 1, endStream = false, endHeaders = true, encodeRequestHeaders(request))
-        sendDATA(streamId = 1, endStream = false, ByteString(0, 0, 0, 0, 0x10, 0x22, 0x0e) ++ ByteString.fromString("GreeterService"))
+        sendDATA(streamId = TheStreamId, endStream = false, ByteString(0, 0, 0, 0, 0x10, 0x22, 0x0e) ++ ByteString.fromString("GreeterService"))
```

Comment:
Why remove the headers here?

Reply:
Because the test setup already includes sending a request. It seems we ignore later HEADERS frames, we need them to support trailers but maybe not in the middle of the stream? Created https://github.com/akka/akka-http/issues/3268 to dig further into this.